File: src/scala/TwoSum.scala
import scala.collection.mutable

object TwoSum {
  def main(args: Array[String]): Unit = {
    twoSum(Array(2,7,11,15),9)
  }

  def twoSum(numbers: Array[Int], target: Int): Array[Int] = {
    val length = numbers.length
    val map = new mutable.HashMap[Int, Int]()
    for (i <- 0 until length) {
      map += (numbers(i) -> i)
    }
    for (i <- 0 until length) {
      val anotherNum = target - numbers(i)
      if (map.contains(anotherNum) && map(anotherNum) != i) {
        return Array[Int](i, map(anotherNum))
      }
    }
    throw new RuntimeException("No such index")
  }
}

File: src/java/org/jy/TwoSum.java
package org.jy;

import java.util.HashMap;
import java.util.Map;

/**
 * 1、两数之和
 *
 * @author LJY
 */
public class TwoSum {

    public int[] twoSum(int[] numbers, int target) {
        int length = numbers.length;
        Map<Integer, Integer> map = new HashMap<>(length);
        for (int i = 0; i < length; i++) {
            map.put(numbers[i], i);
        }
        for (int i = 0; i < length; i++) {
            Integer anotherNum = target - numbers[i];
            if (map.containsKey(anotherNum) && map.get(anotherNum) != i) {
                return new int[]{i, map.get(anotherNum)};
            }
        }
        throw new RuntimeException("No such index");
    }

}
